client/signup/steps/site-options/site-options.tsx

```diff
@@ -32,22 +38,35 @@ const SiteOptions: React.FC< Props > = ( {
 		tagline: defaultTagline,
 	} );
 
+	const [ siteTitleError, setSiteTitleError ] = useState< TranslateResult | null >( null );
+	const [ taglineError, setTaglineError ] = useState< TranslateResult | null >( null );
+
 	const onChange = ( event: React.ChangeEvent< HTMLInputElement > ) => {
 		setFormValues( ( value ) => ( {
 			...value,
 			[ event.target.name ]: event.target.value,
 		} ) );
+		setSiteTitleError( null );
+		setTaglineError( null );
 	};
 
 	const handleSubmit = ( event: React.FormEvent ) => {
 		event.preventDefault();
+		if ( isSiteTitleRequired && ! formValues.siteTitle?.trim().length ) {
+			setSiteTitleError( translate( 'A valid site title is required.' ) );
+			return;
+		}
+		if ( isTaglineRequired && ! formValues.tagline?.trim().length ) {
+			setTaglineError( translate( 'A valid tagline is required.' ) );
+			return;
+		}
 		onSubmit( formValues );
 	};
 
 	return (
 		<form className="site-options__form" onSubmit={ handleSubmit }>
 			<FormFieldset className="site-options__form-fieldset">
-				<FormLabel htmlFor="siteTitle" optional>
+				<FormLabel htmlFor="siteTitle" optional={ ! isSiteTitleRequired }>
```

Comment:
We should also set `require` to `FormLabel`. Besides, I think it's nice to have to disable the submit button if one of the fields is invalid.

Reply:
Adding a `required` attribute to the `FormLabel` would go against the design( uHneAuPrNrodQAvefnzbIA-fi-2716%3A1565), so I skipped it.

Reply:
Okay, but it's weird as the user know this after they click the Continue button and see the error 😂